src/main/java/frc/robot/subsystems/Vision/IOCameraSim.java

```diff
@@ -0,0 +1,86 @@
+package frc.robot.subsystems.Vision;
+
+import edu.wpi.first.math.geometry.*;
+import edu.wpi.first.networktables.*;
+import frc.robot.subsystems.drivetrain.Drive;
+import org.photonvision.SimVisionSystem;
+import org.photonvision.SimVisionTarget;
+
+public class IOCameraSim implements VisionIO {
+    private final NetworkTable table;
+    private final IntegerPublisher publisherInteger;
+    private final DoubleArraySubscriber targetPose;
+    private final DoubleSubscriber targetArea;
+    private final DoubleSubscriber latency;
+    private final DoubleSubscriber targetPitch;
+    private final DoubleSubscriber targetYaw;
+    private final DoubleSubscriber targetSkew;
+    private final IntegerSubscriber targetID;
+    private final BooleanSubscriber hasTarget;
+    SimVisionSystem simVisionSystem;
+    private final Drive swerveDrive = Drive.getInstance();
+
+    public IOCameraSim(int camIndex, String camName) {
+        table = NetworkTableInstance.getDefault().getTable("photonvision").getSubTable(camName);
+
+        publisherInteger = table.getIntegerTopic("pipelineIndex").publish();
+        publisherInteger.setDefault(0);
+
+        simVisionSystem = new SimVisionSystem(
+                camName,
+                90.0,
+                VisionConstants.ROBOT_TO_CAM[camIndex],
+                10000,
+                1600,
+                1200,
+                0.0
+        );
+        for (int i = 0; i < 8; i++) {
+            simVisionSystem.addSimVisionTarget(new SimVisionTarget(aprilChooser(i+1), VisionConstants.TARGET_WIDTH, VisionConstants.TARGET_LENGTH, i+1));
+        }
+        targetPose = table.getDoubleArrayTopic("targetPose").subscribe(new double[7]);
+        targetArea = table.getDoubleTopic("targetArea").subscribe(0);
+        latency = table.getDoubleTopic("latencyMillis").subscribe(0);
+        targetPitch = table.getDoubleTopic("targetPitch").subscribe(0);
+        targetYaw = table.getDoubleTopic("targetYaw").subscribe(0);
+        targetSkew = table.getDoubleTopic("targetSkew").subscribe(0);
+        targetID = table.getIntegerTopic("targetID").subscribe(0);
+        hasTarget = table.getBooleanTopic("hasTarget").subscribe(false);
+    }
+
+    @Override
+    public Pose3d aprilChooser(int aprilID) {
+        return VisionIO.super.aprilChooser(aprilID);
```

Comment:
I don't think that this is the behavior you want (for the usage of super in this class)

Reply:
Well, my intention is that when this method is being called, it will go back to method in the interface and return the required position for the specific aprilID